hyperbahn/advertise_test.go

```diff
@@ -261,48 +261,56 @@ func TestRetryTemporaryFailure(t *testing.T) {
 
 func TestRetryFailure(t *testing.T) {
 	runRetryTest(t, func(r *retryTest) {
-		r.mock.On("On", SendAdvertise).Return().
-			Times(1 /* initial */ + maxAdvertiseFailures /* fail */)
 		r.mock.On("On", Advertised).Return().Once()
+		r.mock.On("On", SendAdvertise).Return().Times(1 + (maxAdvertiseFailures * 2))
+
+		// Since FailStrategyIgnore will keep retrying forever, we need to
+		// signal when we're done testing.
+		doneTesting := make(chan struct{}, 1)
+		// For all but the last failure, we just need to assert that the
+		// OnError handler was called. The callback for the last failure needs
+		// to signal that we're done testing and close the Hyperbahn client (so
+		// that the Advertise call returns).
+		expectationCalled := 0
+		r.mock.On("OnError", ErrAdvertiseFailed{true, advertiseErr}).Return(nil).Times(maxAdvertiseFailures * 2).Run(func(_ mock.Arguments) {
+			expectationCalled++
+			if expectationCalled >= maxAdvertiseFailures*2 {
+				doneTesting <- struct{}{}
```

Comment:
Oh that's right, I forgot about that.

I think the hyperbahn Client.Close being resilient is a good change, so let's keep that.

Looks good, ship it :+1: